test/support/src/helpers.cc

```diff
@@ -1129,8 +1129,14 @@ void write_array(
   tiledb_query_t* query;
   rc = tiledb_query_alloc(ctx, array, TILEDB_WRITE, &query);
   CHECK(rc == TILEDB_OK);
-  if (subarray != nullptr) {
-    rc = tiledb_query_set_subarray(ctx, query, subarray);
+  tiledb_subarray_t* subarray;
+  if (sub != nullptr) {
+    rc = tiledb_subarray_alloc(ctx, array, &subarray);
+    REQUIRE(rc == TILEDB_OK);
+    query->query_->set_subarray(sub);
+    rc = tiledb_subarray_set_subarray(ctx, subarray, sub);
+    CHECK(rc == TILEDB_OK);
```

Comment:
I'd use REQUIRE here, because if this call doesn't succeed, we shouldn't be using it on the query. The test case is going to fail in either case, and _probably_ not with a segfault.